ambry-frontend/src/test/java/com.github.ambry.frontend/AmbryUrlSigningServiceTest.java

```diff
@@ -50,17 +51,56 @@
    * Tests for {@link AmbryUrlSigningServiceFactory}.
    */
   @Test
-  public void factoryTest() {
+  public void factoryTest() throws Exception {
     Properties properties = new Properties();
-    properties.setProperty("frontend.url.signer.upload.endpoint", UPLOAD_ENDPOINT);
-    properties.setProperty("frontend.url.signer.download.endpoint", DOWNLOAD_ENDPOINT);
+    JSONObject jsonObject = new JSONObject().put("POST", UPLOAD_ENDPOINT).put("GET", DOWNLOAD_ENDPOINT);
+    properties.setProperty(FrontendConfig.URL_SIGNER_ENDPOINTS, jsonObject.toString());
     properties.setProperty("frontend.url.signer.default.url.ttl.secs", Long.toString(DEFAULT_URL_TTL_SECS));
     properties.setProperty("frontend.url.signer.default.max.upload.size.bytes", Long.toString(DEFAULT_MAX_UPLOAD_SIZE));
     properties.setProperty("frontend.url.signer.max.url.ttl.secs", Long.toString(MAX_URL_TTL_SECS));
     UrlSigningService signer = new AmbryUrlSigningServiceFactory(new VerifiableProperties(properties),
         new MetricRegistry()).getUrlSigningService();
     assertNotNull("UrlSigningService is null", signer);
     assertTrue("UrlSigningService is AmbryUrlSigningService", signer instanceof AmbryUrlSigningService);
+    assertTrue(((AmbryUrlSigningService) signer).getUploadEndpoint().contains(UPLOAD_ENDPOINT));
+    assertTrue(((AmbryUrlSigningService) signer).getDownloadEndpoint().contains(DOWNLOAD_ENDPOINT));
+  }
+
+  /**
+   * Tests for {@link AmbryUrlSigningServiceFactory}.
+   */
+  @Test
+  public void factoryTestBadJson() throws Exception {
+    Properties properties = new Properties();
+    // Missing GET
+    JSONObject jsonObject = new JSONObject().put("POST", UPLOAD_ENDPOINT);
+    properties.setProperty(FrontendConfig.URL_SIGNER_ENDPOINTS, jsonObject.toString());
+    try {
+      new AmbryUrlSigningServiceFactory(new VerifiableProperties(properties),
+          new MetricRegistry()).getUrlSigningService();
+      fail("Expected exception");
+    } catch (Exception ex) {
+      assertTrue(ex instanceof IllegalStateException);
```

Comment:
you can instead catch only `IllegalStatException`

```
try {
...
} catch (IllegalStateException e) {
  // expected. Nothing to do
}
```
That way, if it is anything other than `IllegalStateException`, we get a stack trace on failure

Reply:
Same comment on all cases in this test

Reply:
Good point